### PR TITLE
feat: Commit consumer offset asynchronously

### PR DESCRIPTION
**Pull request by timmc-edx**
Also add comments, fix lint issue.

**Merge checklist:**
Check off if complete *or* not applicable:
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are squashed
- [x] Noted any: Concerns, dependencies, deadlines, tickets, testing instructions

### CODE REVIEW

File: edx_event_bus_kafka/internal/consumer.py [hunk 1]
```diff
@@ -159,8 +159,12 @@ def consume_indefinitely(self):
                     # to commit all this consumer's current offset across all partitions since we only process one
                     # message at a time, but limit it to just the offset/partition of the specified message
                     # to be super safe
-                    self.consumer.commit(message=msg, asynchronous=False)
+                    #
+                    # Use asynchronous commits for performance; if we used synchronous commits we could
+                    # check the return value for errors, but there's really nothing for us to do with that.
+                    self.consumer.commit(message=msg, asynchronous=True)
```
Comment (timmc-edx) [inline edx_event_bus_kafka/internal/consumer.py:165]: [author's note] `asynchronous=True` is the default, so I could just remove it. Not sure whether to leave it in to be explicit or just to fall back on "Kafka is usually async" as background knowledge.

File: edx_event_bus_kafka/internal/consumer.py [hunk 2]
```diff
@@ -159,8 +159,12 @@ def consume_indefinitely(self):
...
         finally:
+            # Just notifies the broker that this consumer is shutting down. We have auto-commit disabled.
```
Comment (robrap) [inline edx_event_bus_kafka/internal/consumer.py:167]: I think I’d just leave out this comment. Not sure it adds much, and it theoretically could get out of sync.

File: edx_event_bus_kafka/internal/consumer.py (hunk 1 above)
Comment (robrap) [reply edx_event_bus_kafka/internal/consumer.py:165]: Nits:
1. Not sure if any comment is needed, but fine with whatever.
2. I’d be fine with dropping the argument, and just letting the default be.

**Review by robrap — APPROVED** [review]
Nits, but you can decide.

**Comment by timmc-edx** [comment]
OK, I think I'll simplify it.

**Review by robrap — APPROVED** [review]

**Pull request merged**